examples/service_worker/frontend/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+                    {
+                        Ok(true) => None,
+                        Ok(false) => Some(ServiceWorkerError::WorkerUnregistration(None)),
+                        Err(error) => Some(ServiceWorkerError::WorkerUnregistration(Some(error))),
+                    },
+                )
+            });
+        }
+        Msg::WorkerUnregistered(None) => {
+            model.message = Some("Successfully unregistered the service worker.".to_owned());
+        }
+        Msg::WorkerUnregistered(Some(worker_error)) => {
+            model.error = Some(log_error(&worker_error));
+        }
+        Msg::RequestNotificationPermission => {
+            orders.perform_cmd(async {
+                Msg::NotificationPermissionRequested({
+                    web_sys::Notification::request_permission()
+                        .expect("call `request_permission`")
+                        .apply(JsFuture::from)
+                        .await
+                        .map(|js_value| {
+                            web_sys::NotificationPermission::from_js_value(&js_value)
+                                .expect("`NotificationPermission` from `JsValue`")
+                        })
+                        .map_err(ServiceWorkerError::RequestPermission)
+                })
+            });
+        }
+        Msg::NotificationPermissionRequested(Ok(web_sys::NotificationPermission::Granted)) => {
+            let worker_data = match &mut model.worker_data {
+                None => return,
+                Some(worker_data) => worker_data,
+            };
+
+            worker_data.notifications_granted = true;
+
+            // WebPush is a technology that allows a server to send a push event to a service worker. The service worker can then
+            // process the event and perform some action, such as displaying a browser notification. To learn more about WebPush,
+            // see https://medium.com/izettle-engineering/beginners-guide-to-web-push-notifications-using-service-workers-cb3474a17679.
+            // While the above link focuses primarily on javascript, the concepts remain the same. The equivalent rust library can
+            // be found here: https://github.com/pimeys/rust-web-push
+
+            // In order for WebPush to work, the server must have access to a private key when building the signature. The corresponding
+            // public key will be passed when subscribing to the PushManager. A public and private key were generated using openssl for
+            // this example. For reference, here is how both were created:
+            // =======================================
+
+            // Private Key:
+            // openssl ecparam -genkey -name prime256v1 -out private_key.pem
+
+            // Public Key:
+            // openssl ec -in private_key.pem -pubout -outform DER|tail -c 65|base64|tr '/+' '_-'|tr -d '\n'
+            // *Note that I removed the '=' and '%' from the end of the output for the public key
```

Comment:
Without making these changes it will produce the following error:
```
The provided applicationServerKey is not encoded as base64url without padding.
```
See https://github.com/pimeys/rust-web-push#vapid for where I got this command. The nice thing about doing it this way is that we don't have to call `base_64` in the rust code because it's already encoded.

Reply:
I see, it's just a bash alternative to https://github.com/GoogleChromeLabs/web-push-codelab/issues/46#issuecomment-429273981. I think I saw it rewritten in your Rust code in previous PR versions.
So I assume it's just a weird protocol requirement or something like that.

Add some comments about it with links if you think it makes sense for future readers.